x-pack/solutions/observability/plugins/synthetics/server/routes/monitor_cruds/project_monitor/add_monitor_project.ts

```diff
@@ -109,21 +113,91 @@ export const addSyntheticsProjectMonitorRoute: SyntheticsRestApiRouteFactory = (
   },
 });
 
-export const REQUEST_TOO_LARGE = i18n.translate('xpack.synthetics.server.project.delete.request', {
-  defaultMessage:
-    'Request payload is too large. Please send a max of 250 browser monitors per request.',
-});
+const validProjectMultiSpace = async (routeContext: RouteContext, monitors: ProjectMonitor[]) => {
+  const { response } = routeContext;
+  const spaceId = await validateSpaceId(routeContext);
 
-export const REQUEST_TOO_LARGE_LIGHTWEIGHT = i18n.translate(
-  'xpack.synthetics.server.project.delete.request.lightweight',
-  {
-    defaultMessage:
-      'Request payload is too large. Please send a max of 1500 lightweight monitors per request.',
+  const spacesList = new Set(
+    monitors
+      .map((monitor) => monitor.spaces ?? [])
+      .flat()
+      .filter((sp) => sp !== ALL_SPACES_ID)
+  );
+  if ((spacesList.size === 1 && spacesList.has(DEFAULT_SPACE_ID)) || spacesList.size === 0) {
+    return spaceId;
   }
-);
+
+  try {
+    await pMap(
+      spacesList,
+      async (space) => {
+        const { server } = routeContext;
+        const spacesClient = server.spaces?.spacesService.createSpacesClient(routeContext.request);
```

Comment:
We can create the `spacesClient` outside of this function, no need to recreate it for each space.